### DISCUSSION TITLE
Problem in fitting the model

### DISCUSSION THREAD

**Discussion by Mahdi22223** [georgebv/pyextremes#51] [Q&A] [answered]
Dear Valuable team,

I am a new user of pyextremes package which i need to fit my data which is timeseries of tension loads on mooring line. lets say (t,x), and i drived them from FEM analysis. So first i transffered to the series format as suggeted and then storred them in data.frame like this test = pd.Series(data =df['x'].values, index = df['t']). Then i tried to use the quick example of the package. these two codes are running sucesfuklly: model = EVA(data = test)
model.get_extremes(method="BM", block_size="365.2425D")
however when i tried to  go on next step and run model.fit_model(), the package give me error as:

model.fit_model()
Traceback (most recent call last):

  Cell In[18], line 1
    model.fit_model()

  File ~\anaconda3\lib\site-packages\pyextremes\eva.py:601 in fit_model
    candidate_models = {

  File ~\anaconda3\lib\site-packages\pyextremes\eva.py:602 in <dictcomp>
    distribution_name: MLE(

  File ~\anaconda3\lib\site-packages\pyextremes\models\model_mle.py:43 in __init__
    super().__init__(

  File ~\anaconda3\lib\site-packages\pyextremes\models\model_base.py:65 in __init__
    self.distribution = Distribution(

  File ~\anaconda3\lib\site-packages\pyextremes\models\distribution.py:110 in __init__
    self.mle_parameters = self.fit(data=self.extremes.values)

  File ~\anaconda3\lib\site-packages\pyextremes\models\distribution.py:134 in fit
    full_mle = self.distribution.fit(data=data, **self.fixed_parameters)

  File ~\anaconda3\lib\site-packages\scipy\stats\_distn_infrastructure.py:2655 in fit
    if not np.isfinite(data).all():

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

I will be thankful if you kindly advice me on this problem.

**Comment by georgebv** [comment]
You should check your data for invalid values. What dtype are you using?

**Reply by Mahdi22223** [reply]
Dear georgebv, thank you so much to getting back to me. You are correct. It was a problem with my time series. since my data are based on second with time step of 0.1, The tensions are also in KN all are numeric values. The total time seris are only for 50 minutes. I did the same processas the quick guid example and fortunately i get the result sucessfully. I used this code according to my data:

time = line1.SampleTimes(period)
time = time/60

t = pd.to_datetime(time, unit ='m')
y = pd.Series(tension1)

df = pd.DataFrame({'t':t, 'y':y})

test = pd.Series(data =df['y'].values, index = df['t'])

series = (
    series
    .sort_index(ascending=True)
    .astype(float)
    .dropna()
)
series = series - (series.index - series.index[0]) / pd.to_timedelta("1Min") * 2.87e-3

model = EVA(data = series)
model.get_extremes(method="BM", block_size="1Min")
model.plot_extremes()


model.fit_model()

summary = model.get_summary(
    return_period=[1, 2, 5, 10, 25, 50, 100, 250, 500, 1000],
    alpha=0.95,
    n_samples=1000,
)

model.plot_diagnostic(alpha=0.95)

Since my data are in minutes then the plot did not predict any return period, which is remain at 0. Is there any option with model.fit_model() that i can tell the pyextremes to fit my data according to minutes and predic the return period based on minutes. I attached the picture as well.
![Plot_Exterem](https://user-images.githubusercontent.com/94902954/234484654-6b3061f8-cf83-4688-b92e-020cf58e9ef3.png)

**Reply by georgebv** [reply, marked as answer]
You can use any units, the default period is 365 days. You should set your period both when extracting extremes (as you did) and when generating figures/results. Fitting doesn't care about your time scale - you fit distributions to a sample of extremes, not to time series (you can see that in the PDF plot).

**Reply by Mahdi22223** [reply]
Dear George, Thank you so much. It works now.
I tried to shift to test POT on my data, but first tried to use the quick example again to make sure i am following the same role. I am using the latest version of python and everything is updated. when i used the code for POT is gave me the error that **Cell In[63], line 5
    plot_extremes(
TypeError: plot_extremes() got an unexpected keyword argument 'r'**

However, from the help for plot_extremes()  function said that for POT i must define the "r" as well. I am not sure if this error happend becsue of my python software, or there is somthing wrong in the code i used. I checked it manytimes, but could not find the issue. just note that when i remove "r" from the code then the python respond to the code. It seesm that it cannot define r with the extreme function in my python, but everthing is going well with BM method. I will be thankful if you kindly advice me on this

**Reply by Mahdi22223** [reply]
Dear George, the code is now working. I understand that it was a bug from my python, when i remove spaces then the code works!!
by the way, I am curious to know that why BM method gives better result compare to POT. I was expecting vis versa.[
![BM](https://user-images.githubusercontent.com/94902954/234789977-77309c14-c58d-418b-969c-ffd65a322855.png)
![POT](https://user-images.githubusercontent.com/94902954/234789989-8c23f2f6-591b-4aa4-8767-8a69c0b3ce73.png)
](url)
